app/lib/preload.js

```diff
@@ -9,29 +9,82 @@
  */
 
 const {
-  remote,
+  contextBridge,
   ipcRenderer
 } = require('electron');
 
-const {
-  process,
-  app
-} = remote;
+const allowedEvents = [
+  'errorTracking:turnedOn',
+  'errorTracking:turnedOff',
+  'external:open-url',
+  'dialog:open-files',
+  'dialog:open-file-error',
+  'dialog:save-file',
+  'dialog:show',
+  'file:read',
+  'file:read-stats',
+  'file:write',
+  'zeebe:checkConnection',
+  'zeebe:deploy',
+  'zeebe:run',
+  'config:get',
+  'config:set',
+  'toggle-plugins',
+  'client:ready',
+  'client:error',
+  'app:quit-allowed',
+  'workspace:restore',
+  'workspace:save',
+  'menu:register',
+  'menu:update',
+  'context-menu:open'
+];
 
-const {
-  platform
-} = process;
+const api = {
+  send: (event, ...args) => {
+    if (!allowedEvents.includes(event)) {
+      throw new Error(`Disallowed event: ${event}`);
+    }
+
+    ipcRenderer.send(event, ...args);
+  },
+  on: (event, callback) => {
+    ipcRenderer.on(event, callback);
+
+    return {
+      cancel() {
+        ipcRenderer.off(event, callback);
+      }
+    };
+  },
+  once: (event, callback) => {
+    ipcRenderer.once(event, callback);
+
+    return {
+      cancel() {
+        ipcRenderer.off(event, callback);
+      }
+    };
+  }
+};
+
+let executed = false;
+
+contextBridge.exposeInMainWorld('getAppPreload', function() {
 
-/* global window */
+  // expose api only once
+  // related to https://github.com/camunda/camunda-modeler/issues/2143
+  if (executed) {
+    return;
```

Comment:
Could be helpful to blow up here, so we do not silently discard miss-use.

Reply:
👍

Reply:
```suggestion
    throw new Error('🚔');
```

Reply:
`throw new Error('this is recorded! We'll track you down!!!!11^^10815')`.